### ISSUE TITLE
Why not on safari ios?

### ISSUE THREAD

**Issue by tobiasgessler** [Novage/p2p-media-loader#32]
Why is it not possible to deliver hls to ios devices?
They do support WebRTC Data Channels and I don't really understand what's the problem.
I also think that peer5 is able deliver hls to ios devices vie peer to peer but I wasn't able to figure out how.
Is it pollible to do so and just not supported by this p2p-media-loader or is it impossible to deliver hls to ios devices over WebRtc?

**Comment by mrlika** [comment]
You are right - WebRTC Data Channels work on iOS Safari but it doesn't support Media Source Extensions API. It means that we can download media over P2P but can't play it. You can't put what you've  downloaded into <video> element without Media Source Extensions API.

Moreover, iOS Safari supports only HLS and only natively as a source URL for <video> element. You can't even implement bitrate switch controls for it without hacks.

> I also think that peer5 is able deliver hls to ios devices vie peer to peer but I wasn't able to figure out how.

Only in applications. They run proxy server that do media over P2P in the background. This approach doesn't work in iOS Safari directly.

> Is it pollible to do so and just not supported by this p2p-media-loader or is it impossible to deliver hls to ios devices over WebRtc?

Exists one trick that can make it possible. As I know currently not implemented by other P2P solutions. We are working on it but still even no prototype.
